krm-functions/dnn-fn/fn/function.go

```diff
@@ -190,8 +190,10 @@ func (f *dnnFn) updateDnnResource(dnnObj_ *fn.KubeObject, owned fn.KubeObjects)
 
 func getAnnotations(annotations map[string]string) map[string]string {
 	a := map[string]string{}
-	for k, v := range annotations {
-		a[k] = v
+	if len(annotations) > 0 {
```

Comment:
this shouldn't be needed, range should handle nil and zero length just fine. Why was this necessary?

Reply:
I did this change and nothing worked any longer so was a bit paranoid and added this, but it did not change anything. So will revert

Reply:
been there, done that :)